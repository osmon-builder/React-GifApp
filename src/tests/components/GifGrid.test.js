import React from 'react';
import '@testing-library/jest-dom';
import {shallow} from 'enzyme';
import { GifGrid } from '../../Componentes/GifGrid';
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('GrifGridu', () => {

    const category = 'Goku';
    

    test('Mostrar el componente correctamente ', () => {

        useFetchGifs.mockReturnValue({
            data: [],
            loading: true
        });

        const wrapper = shallow(<GifGrid category = {category}/>)
        expect(wrapper).toMatchSnapshot();
    });

    test('Debe de mostrar items cuando se cargan imagenes useFetchGifs', () => {

        const gifs = [{
            id: 'ABC',
            url: 'http://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
        },
            {id: '123',
            url: 'http://localhost/cualquier/cosa.jpg',
            title: 'Cualquier cosa'
            }   
        ];

        useFetchGifs.mockReturnValue({
            data: gifs,
            loading: false
        });
       
        const wrapper = shallow(<GifGrid category = {category}/>)

        
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('p').exists()).toBe(false);
        expect(wrapper.find('GifGridItem').length).toBe( gifs.length );
        
    })
    
    
   

   
  
    
})
