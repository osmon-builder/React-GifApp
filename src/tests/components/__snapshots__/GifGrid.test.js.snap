// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`GrifGridu Debe de mostrar items cuando se cargan imagenes useFetchGifs 1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
     
    Goku
     
  </h3>
  <div
    className="card-grid"
  >
    <GifGridItem
      id="ABC"
      key="ABC"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
    <GifGridItem
      id="123"
      key="123"
      title="Cualquier cosa"
      url="http://localhost/cualquier/cosa.jpg"
    />
  </div>
</Fragment>
`;

exports[`GrifGridu Mostrar el componente correctamente  1`] = `
<Fragment>
  <h3
    className="animate__animated animate__bounce"
  >
     
    Goku
     
  </h3>
  <p
    className="animate__animated animate__flash"
  >
    loading
  </p>
  <div
    className="card-grid"
  />
</Fragment>
`;
